packages/office-ui-fabric-react/src/components/Nav/Nav.types.ts

```diff
@@ -154,6 +154,11 @@ export interface INavLinkGroup {
    * ARIA label when group is collapsed and can be expanded.
    */
   collapseAriaLabel?: string;
+
+  /**
+   * (Optional) Any additional properties to apply to a group.
+   */
+  [propertyName: string]: any;
```

Comment:
Would you mind adding some examples of scenarios this enables? Prefer to avoid opening up the types with an `any` like this.

Reply:
I would advise against allowing direct attached props, in favor of forcing them through a `data?: any` prop, similar to what we've done with other components. Adding an indexer winds up causing TypeScript to degrade the type to `any`, which can mask typos and other bugs when using this interface.

Reply:
Agreed I prefer not having to use `any` as well, I have updated it to groupData @ThomasMichon 

@jdhuntington An example scenario would be attaching additional props for user later on in the onRender function of a group header.